src/array.ts

```diff
@@ -1,31 +1,27 @@
-export const objectListForAttribute = <T>(
+export const groupByKey = <T>(
   objects: T[],
-  attribute: string
+  key: string
 ): Record<string, T[]> => {
-  return objects.reduce(
-    (objectListForAttribute: Record<string, T[]>, object: T) => {
-      objectListForAttribute[object[attribute]]
-        ? objectListForAttribute[object[attribute]].push(object)
-        : (objectListForAttribute[object[attribute]] = [object]);
-      return objectListForAttribute;
-    },
-    {}
-  );
+  return objects.reduce((grouped: Record<string, T[]>, object: T) => {
+    const keyValue = object[key];
```

Comment:
I typically try not to drill further than one value so I usually destructure or assign a const.